src/versionize/Cargo.toml

```diff
@@ -10,6 +10,7 @@ bincode = "1.2.1"
 versionize_derive = { path = "../versionize_derive" }
 crc64 = "1.0.0"
 vmm-sys-util = ">=0.4.0"
+vm-memory = { version = ">=0.2.0" }
```

Comment:
I don't agree that the `versionize` crate should depend on abstractions of the guest memory. I think it shouldn't care about the guest memory at all. It defines serialization for structs. It doesn't even need to be virtualization related. When versioned serialization is available in `rust-vmm`, we'll have to figure out if `vm-memory` needs to pull it in as a dependency and learn to serialize its `GuestAddress`es. Personally I think it shouldn't. Anyway that's a discussion for later.

In the meantime, since we can't `impl Versionize for GuestAddress` in the VMM (not the `versionize` crate) and we don't support any mechanism like `versionize_this_unversionizable_member_with()`, I think we should shadow the structs that have un-versionizable members with "state" structs where said members are broken down into either nested state, or primitives.

For example, the [`virtio-net` device](https://github.com/firecracker-microvm/firecracker/blob/0b7ba621d4c8a3f8948e7ebce3421f0262208eba/src/devices/src/virtio/net/device.rs#L60) contains a [`Tap`](https://github.com/firecracker-microvm/firecracker/blob/0b7ba621d4c8a3f8948e7ebce3421f0262208eba/src/utils/src/net/tap.rs#L49), which in turns contains a `File`, which we won't save. We'll only save the tap's name in a separate struct.

Reply:
I agree that the dependency shouldn't be there, but I believe we would end up shadowing a lot of structures and that would imply copying the fields by hand from one struct to another. Also this would also make it impossible to nest such a struct into a struct that fully supports Versionize on all fields.

I think for the short term it makes sense to at least try to refactor the state structures so we do not end up implementing Versionize for other other crate structs and otherwise fall back  to the currently implemented solution. We can incrementally fix this once we decide on a good solution for versioning `remote` crate structs.

For the net device we will need to modify the device constructor to receive the Path also that we will serialize and keep the File outside the state structure.